### PR TITLE
Service labels

### PR DESCRIPTION
**Pull request by tegataiprime**
Adjusted MarathonDiscoveryClient to return Marathon Application Labels (a map of key/value pairs) as the Spring Cloud Discovery ServiceInstance meta-data (also a map of key/value pairs).  This will facilitate applying Predicate Filters using Ribbon e.g. "fetch all service instances with the following key/value combinations".  This enables us to also announce service capabilities.

### CODE REVIEW

**Comment by aatarasoff** [comment]
Nice feature. 
Would you like to provide possible usage in example project and documentation?

**Comment by tegataiprime** [comment]
Yes, of course.  I am working on another complimentary feature to locate services using a regular expression or wild card.  Then applying a ribbon filter over ServiceInstance meta-data makes a lot more sense, as each service located could have a different set of labels.  This kind of thing really helps when applying rolling updates.  I'll keep working on this branch & will update the documentation & example application.

**Comment by tegataiprime** [comment]
Note: depends upon com.mesosphere:marathon-client:0.3.1-SNAPSHOT (to support DC/OS secrets that are returned in App.env)

File: autoconfigure/src/main/java/mesosphere/marathon/client/Marathon.java [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+import feign.Param;
+import feign.RequestLine;
+
+public interface Marathon {
+    // Apps
+    @RequestLine("GET /v2/apps")
+    GetAppsResponse getApps() throws MarathonException;
+
+    @RequestLine("GET /v2/apps")
+    GetAppsResponse getApps(@QueryMap Map<String, String> queryMap) throws MarathonException;
```
Comment (tegataiprime) [inline autoconfigure/src/main/java/mesosphere/marathon/client/Marathon.java]: Added a new method to support passing query parameters to /v2/apps
e.g.
/v2/apps?id=myServiceName

Submitted a PR to mesosphere/marathon-client for this change:
https://github.com/mesosphere/marathon-client/pull/44

File: build.gradle [hunk 2]
```diff
@@ -22,6 +22,8 @@ allprojects {
 
     repositories {
         jcenter()
+        mavenLocal()
```
Comment (tegataiprime) [inline build.gradle:25]: This is temporary.  Have built com.mesosphere:marathon-client:0.3.1-SNAPSHOT locally.

File: build.gradle [hunk 3]
```diff
@@ -40,7 +42,7 @@ allprojects {
 
         dependencies {
             dependency 'org.projectlombok:lombok:1.16.8'
-            dependency 'com.mesosphere:marathon-client:0.3.0'
+            dependency 'com.mesosphere:marathon-client:0.3.1-SNAPSHOT'
```
Comment (tegataiprime) [inline build.gradle:45]: Using com.mesosphere:marathon-client:0.3.1-SNAPSHOT (built locally) to support DC/OS secrets in App.env

File: build.gradle (hunk 2 above)
Comment (aatarasoff) [reply build.gradle:25]: I think we could not wait until new Marathon lib version will be released, cause you've already patched Marathon client. So, remove that for passing travis tests and I merge your PR.

File: build.gradle (hunk 3 above)
Comment (aatarasoff) [reply build.gradle:45]: See my another comment about maven local.

**Comment by aatarasoff** [comment]
There are some troubles with PR after merging. 
Maybe you not commit some files like DCOSClient and etc?

Could you fix the problem?

**Comment by aatarasoff** [comment]
Anyway, I've eliminated DCOS part, reverted marathon client lib version to 0.3.0 and fixed build.

Thank for you pull request. If you want DCOS part to comeback, please, merge with master and make new pull request with changes that will pass CI build.

**Comment by aatarasoff** [comment]
Do you have twitter? I wanna mention you in tweet about new version of starter.

**Comment by tegataiprime** [comment]
Yes.  My twitter handle is @tegatai

**Comment by tegataiprime** [comment]
I'm talking with the chaps at Mesosphere about getting my PR for QueryParms merged in and to have them release their 0.3.1 version.  We have a commercial relationship with them, so I am hopeful this will happen soon.

When that is done I'll be able to re-commit the changes for DCOS authentication.  In the meantime, I've created my own artifacts for marathon-client:0.3.1 & spring-cloud-marathon (with DCOS authentication) to keep our internal development moving forward.

**Comment by aatarasoff** [comment]
Ok, that's great!

**Comment by tegataiprime** [comment]
My changes to marathon-client have been accepted & merged into their master.  Waiting on Mesosphere for a release of 0.3.1.

**Comment by tegataiprime** [comment]
Mesosphere are preparing to release marathon-client 0.5.0, which contains my changes.  When it hits maven central I'll open a PR to merge in support for DCOS authentication.

**Comment by aatarasoff** [comment]
Ok, it is great news. Will be waiting for changes)

**Pull request merged**